src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -493,7 +493,10 @@ private Dataset retrieveWorker(DatasetSummary summary) {
         // Retrieve the project and application deployment resource associated with the dataset
         // This is a bit sketchy filling in the object via a dao in another package.
         // It seemed like the cleanest thing to me at the time.
-        dataset.projectResource(resourceService.getProjectResource(dataset.getProjectResourceId()));
+        if (dataset.getProjectResourceId() != null) {
```

Comment:
I'm not too familiar with this code, but is there a reason to keep this field around once this change is complete? Are there existing azure datasets in TDR that we need to support for backward compatibility reasons? I would expect to see the field go away completely.